pipeline_steps/common.groovy

```diff
@@ -354,6 +354,16 @@ def prepareConfigs(Map args){
 
 def prepareRpcGit(Map args){
   dir("/opt/rpc-openstack"){
+
+    /* if job is triggered by PR, then we need to set RPC_REPO and
+       RPC_BRANCH using the env vars supplied by ghprb.
+    */
+    if ( env.ghprbPullId != null ){
+      env.RPC_REPO = "https://github.com/${env.ghprbGhRepository}.git"
+      env.RPC_BRANCH = "origin/pr/${env.ghprbPullId}/merge"
```

Comment:
This isn't used in the checkout below, args.branch is. Also we still need to be able to override the branch for checking out the upgrade_from version. 

How about:
```
def prepareRpcGit(branch="auto"){
  if (branch == "auto"){
    if (env.ghprbPullId != null){
      env.RPC_REPO = "https://github.com/${env.ghprbGhRepository}.git"
      branch = "origin/pr/${env.ghprbPullId}/merge"
    }
    branch = env.RPC_BRANCH
  }

 ...

```
Note that `args.branch` will need to change to `branch` and `branch: env.RPC_BRANCH` will need to be removed from all the calls to `prepareRpcGit`. The upgrade call will be `common.prepareRpcGit(env.UPGRADE_FROM_REF)`